src/common/procedure/src/error.rs

```diff
@@ -0,0 +1,85 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::any::Any;
+
+use common_error::prelude::*;
+
+use crate::procedure::ProcedureId;
+
+/// Procedure error.
+#[derive(Debug, Snafu)]
+#[snafu(visibility(pub))]
+pub enum Error {
+    #[snafu(display(
+        "Failed to execute procedure due to external error, source: {}",
+        source
+    ))]
+    External {
+        #[snafu(backtrace)]
+        source: BoxedError,
+    },
+
+    #[snafu(display("Loader {} is already registered", name))]
+    LoaderConflict { name: String, backtrace: Backtrace },
+
+    #[snafu(display("Failed to serialize to json, source: {}", source))]
+    ToJson {
+        source: serde_json::Error,
+        backtrace: Backtrace,
+    },
+
+    #[snafu(display("Procedure {} already exists", procedure_id))]
+    DuplicateProcedure {
+        procedure_id: ProcedureId,
+        backtrace: Backtrace,
+    },
+
+    // Currently we don't provide backtrace for an invalid ProcedureId as
+    // this error is mainly for the procedure manager to check whether a
+    // key in the procedure store is valid.
+    #[snafu(display("Invalid procedure id, source: {}", source))]
+    InvalidId { source: uuid::Error },
```

Comment:
Also add backtrace to `InvalidId`?

Reply:
In fact, the backtrace is useless as the procedure manager usually uses this to test whether a path is valid and skip that path if it doesn't contain a procedure id.

I think I can define a dedicated error for this like RecvError of channel.

Reply:
In fact, the backtrace is useless as the procedure manager usually uses this to test whether a path is valid and skip that path if it doesn't contain a procedure id.

I think I can define a dedicated error for this like RecvError of channel.